vmnics.go

```diff
@@ -0,0 +1,185 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"strings"
+
+	"github.com/apex/log"
+	"github.com/pkg/errors"
+)
+
+const randChars = "abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789"
+
+func randomString(length int) string {
+	if length == 0 {
+		length = 8
+	}
+	randCharLength := len(randChars)
+	b := make([]byte, length)
+	for i := range b {
+		b[i] = randChars[rand.Intn(randCharLength)]
+	}
+	return string(b)
+}
+
+const LinuxIfnameLength int = 15
+
+// generate a random Linux network interface name.
+// Allow specifying a prefix
+func randomNicIFName(prefix string) string {
+	prefixLength := len(prefix)
+	if prefixLength >= LinuxIfnameLength {
```

Comment:
Shall we go a step further and require at least space for two random characters?  The pain of debugging subtle breakage when ifnames become likely to be the same is probably going to be worse than that of anyone wanting to use 14 character prefix.

Reply:
Sure